File: TerminBA/TerminBA.UI/terminba_mobile/lib/model/cancellation_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'cancellation_response.g.dart';

@JsonSerializable()
class CancellationResponse {
  final String reservationState;
  final bool refundIssued;
  final String? refundStatus;
  final double? refundAmount;

  CancellationResponse({
    required this.reservationState,
    required this.refundIssued,
    this.refundStatus,
    this.refundAmount,
  });

    factory CancellationResponse.fromJson(Map<String, dynamic> json) =>
      _$CancellationResponseFromJson(json);
  Map<String, dynamic> toJson() => _$CancellationResponseToJson(this);
}

File: TerminBA/TerminBA.Models/Model/CancellationResponse.cs
namespace TerminBA.Models.Model
{
    public class CancellationResponse
    {
        public string ReservationState { get; set; } = string.Empty;
        public bool RefundIssued { get; set; }
        public string? RefundStatus { get; set; }
        public decimal? RefundAmount { get; set; }
    }
}
